packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -602,14 +605,43 @@ export function TransferPanel() {
         isBatchTransferSupported &&
         Number(amount2) > 0
 
+      let approvalAmountIncrease: BigNumber | undefined = undefined
+
+      // Eth transfers to a custom destination use retryables
+      // In the case of native currency we need to also approve native currency used for gas
+      if (!selectedToken && isCustomDestinationTransfer) {
+        const retryableGasEstimates =
+          await bridgeTransferStarter.transferEstimateGas({
+            amount: amountBigNumber,
+            signer
+          })
+
+        const parentGasPrice = await parentChainProvider.getGasPrice()
+
+        const parentRetryableGas =
+          retryableGasEstimates?.estimatedParentChainGas.mul(parentGasPrice)
+
+        if (parentRetryableGas) {
+          approvalAmountIncrease = percentIncrease(
+            parentRetryableGas,
+            BigNumber.from(30)
+          )
+        }
+      }
+
+      if (isCustomNativeTokenAmount2) {
+        approvalAmountIncrease = utils.parseUnits(
+          amount2,
+          nativeCurrency.decimals
+        )
+      }
```

Comment:
in `EthDepositStarter`, `approveNativeCurrency` can call `isCustomDestinationAddressTx` to determine if the retryable gas needs to be added to the approval amount?

It doesn't feel right to have this code here

but if we must, perhaps we can use do sth like this outside the `transfer` function? It is not ideal to grow the number of lines within the `transfer` function

hook file
```typescript
export function useApprovalAmountIncrease() {
  return useMemo(() => {
  }, [deps])
}
```

TransferPanel.tsx
```typescript
const approvalAmountIncrease = useLatest(useApprovalAmountIncrease())
```

Reply:
don't know about the hook, doesn't feel like a suitable thing for a hook